### PR TITLE
Deprecate MySQL < 5.6, MariaDB < 5.5

### PR DESCRIPTION
**Pull request by yol**
We do not have the resources to maintain compatibility with all sorts of
ancient MySQL servers in the future (concerning both development
manpower and testing infrastructure). Therefore, deprecate very old
MySQL and MariaDB server versions so that we can remove support for them
when problems come up or we would have to go out of our way to keep them
happy.

### CODE REVIEW

**Comment by pkerling** [comment]
@MartijnKaijser what do you think?

**Comment by MartijnKaijser** [comment]
Does it still work or should we bail out gracefully (just not start and throw an error)?

**Comment by pkerling** [comment]
At the moment, we should be fine (since we already went through a few iterations and fixed stuff for <5.7 specifically). I just want to get this in so we don't have to do that again in the future and people are aware that they are running ancient stuff. If we do break compatibility (by accident or on purpose) at some point, I think we should bail out.

**Comment by MartijnKaijser** [comment]
We might even need to go further than log only and show a GUI pop-up imo.

**Comment by pkerling** [comment]
IMO that might be a bit too noisy (on every start?) and... I just have no idea how to even do that from that part of the code :see_no_evil: MySQL DB is already an advanced feature without GUI setup so I kind of assume that people running such a setup are more "prone" to scroll through the logs :-)

**Comment by MartijnKaijser** [comment]
Let's merge this as a start

File: xbmc/dbwrappers/mysqldataset.cpp [hunk 1]
```diff
@@ -176,8 +176,22 @@ int MysqlDatabase::connect(bool create_new) {
       static bool showed_ver_info = false;
       if (!showed_ver_info)
       {
-        CLog::Log(LOGINFO, "MYSQL: Connected to version %s", mysql_get_server_info(conn));
+        std::string version_string = mysql_get_server_info(conn);
+        CLog::Log(LOGINFO, "MYSQL: Connected to version {}", version_string);
```
Comment (MilhouseVH) [inline xbmc/dbwrappers/mysqldataset.cpp]: Maybe this could be LOGNOTICE so that it is logged even when debug is not enabled?

**Comment by pkerling** [comment]
updated

File: xbmc/dbwrappers/mysqldataset.cpp (hunk 1 above)
Comment (pkerling) [reply xbmc/dbwrappers/mysqldataset.cpp]: fine by me

**Comment by thezoggy** [comment]
https://www.fromdual.com/support-for-mysql-from-oracle
mysql 5.5 came out on Dec 2010 and its extended support ended on Dec 2018. so probably good time to depreciate support for it.

**Comment by pkerling** [comment]
@thezoggy I agree, but team opinion was not with me. Strongest argument in my opinion: Debian stable still ships mysql-server-5.5 and ARM users cannot use the newer mysql.com packages since they only provide x86/x64.

**Comment by Rechi** [comment]
> Debian stable still ships mysql-server-5.5

That's only partly correct. Debian stable ships mariadb 10.1.37 as default mysql server, https://packages.debian.org/stable/default-mysql-server.

**Comment by pkerling** [comment]
> Edit: Not even partly correct. Debian stable (stretch) only ships mariadb, no mysql package. The packages having mysql in it's name are only pseudo packages depending on mariadb packages.

You are right. @popcornmix?

**Comment by pkerling** [comment]
For context: The bottom line of the team discussion was that there is no need to deprecate versions that are still in widespread use (which 5.5 arguably is) as long as there are still people willing to do the required testing (@MilhouseVH) and fixing (@DaveTBlake).

**Comment by MartijnKaijser** [comment]
Regardless there's no harm in putting a notice in the log to ask users to upgrade to something newer if available.

**Comment by pkerling** [comment]
I think that warning for MySQL <5.6 (because 5.5 is EOL) and MariaDB <5.5 (RHEL is on 5.5 so no 10.1 unfortunately, and for MariaDB, 5.5 is still supported)is a fair compromise, esp. given that we know now that Debian stable fulfils this. @MilhouseVH @DaveTBlake @popcornmix ?

**Comment by popcornmix** [comment]
I'm okay with that.

**Comment by DaveTBlake** [comment]
Sure, warn away.

**Pull request merged**